drafts/maui-test_pt1.md

```diff
@@ -0,0 +1,332 @@
+---
+title: How to test a .NET MAUI App Part 1
+domain: software-engineering-corner.zuehlke.com
+tags: dotnet,mvvm,xaml,net-maui,csharp,xunit,testing,moq
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1715274155168/4q4lrfJcY.png?auto=format
+publishAs: NiZarper
+hideFromHashnodeCommunity: false
+saveAsDraft: true
+---
+
+# How to test a .NET MAUI App Part 1
+
+This blog post is the first part of a two-part series on testing a .NET MAUI application. This first post focuses on the service and view model layers, while the second will focus on the UI testing aspect.
+
+## Table Of Contents
+
+- Setup
+- Testing of a service
+- Testing of a view model
+- Takeaways
+
+## Setup
+
+This chapter provides a practical guide to the setup process for testing a .NET MAUI project.
+
+1. Add a new test project. In my case, I used XUnit.
```

Comment:
Depending on what you want to convey, you might want to think about changing all instances of "I" with "we". Even though it is very likely that you actually did this on your very own without help and resources from others, that fact might be irrelevant here. But maybe this can be discussed with someone from Marketing, what it is that is to be conveyed. I'm unsure and would approach them.

Reply:
This is a tough one. I know that we write from the company's perspective, but we also need to consider that there are posts, like mine, that are based on hobby learning projects. In addition, I checked some posts of our colleagues, and there I observed that the perspective used depends on multiple factors, e.g., the presented approach was used in a customer project, the author reviewed a new technology purely out of interest, etc.

In my case, I prefer to stay with "I" because all my previous blog posts have been based on the same hobby learning project.

Reply:
Use whatever fits the context / type of article. As you said, if it's about patterns/technologies/methods we regularly use and apply in customer projects, 100% go with "we". In guides and walkthroughs "we" could also be applied, referring to author _and_ reader, or a larger community (web devs, mobile engineers, technology sector, …).

In general, I would only use "I" with personal opinions, or experiences on private side-projects/training. But we clearly haven't been very consistent with this so far.

For this specific article, I think "we" could work very nicely. A lot of it is already (e.g. "let us", "we can centralize …"). The rest could be changed like: "Add a new test project. In this article, we will use XUnit."

Reply:
Use whatever fits the context / type of article. As you said, if it's about patterns/technologies/methods we regularly use and apply in customer projects, 100% go with "we". In guides and walkthroughs "we" could also be applied, referring to author and reader, or a larger community (web devs, mobile engineers, technology sector, …).

In general, I would only use "I" with personal opinions, or experiences on private side-projects/training. But we clearly haven't been very consistent with this so far.

For this specific article, I think "we" could work very nicely. A lot of it is already (e.g. "let us", "we can centralize …"). The rest could be changed like: "Add a new test project. In this article, we will use XUnit."

Reply:
Okay, that sounds plausible to me. I changed the personal pronoun from I to we where applicable. I suggest we document such decisions inside a wiki or readme for the future to keep our posts consistent.